fms/fms_io.F90

```diff
@@ -7909,7 +7911,11 @@ subroutine get_mosaic_tile_grid(grid_file, mosaic_file, domain, tile_count)
     allocate(tile_id(ntileMe))
     tile_id = mpp_get_tile_id(domain)
     call read_data(mosaic_file, "gridfiles", grid_file, level=tile_id(tile) )
-    grid_file = 'INPUT/'//trim(grid_file)
+    if (.not. present(custom_path)) then
+      grid_file = 'INPUT/'//trim(grid_file)
+    else
+      grid_file = trim(custom_path)//'/'//trim(grid_file)
+    endif
```

Comment:
There are entries in the grid_spec.nc for the <comp>_mosaic_dir, as well as the 'gridlocation' within the individual grid mosaics.  If one does not use relative paths that are specific to the rundir directory structure, these cannot be used.  However, there are NetCDF file manipulation tools that one could use during job creation to allow us to support this option as an alternative to the custom pathing option.  Thoughts?

Reply:
Would there be an issue with doing it this way? I fear that adding these extra steps into the workflow would add quite a bit of complexity. For example you don't have to run with these grid spec files, in most cases we create the grid on the fly. It's only the regional system that needs to read the grid. So the workflow would have to have extra conditionals on that and it could get rather complicated.

Reply:
No, I was merely pointing out the capability for discussion.  As it seems to add complexity, I am okay with your proposed solution.

Reply:
OK, thanks. It's good to know about the other solution.